enterprise/server/remote_execution/containers/podman/podman.go

```diff
@@ -0,0 +1,157 @@
+package podman
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/commandutil"
+	"github.com/buildbuddy-io/buildbuddy/enterprise/server/remote_execution/container"
+	"github.com/buildbuddy-io/buildbuddy/server/interfaces"
+	"github.com/buildbuddy-io/buildbuddy/server/util/background"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/buildbuddy-io/buildbuddy/server/util/random"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+
+	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
+)
+
+var (
+	containerFinalizationTimeout = 10 * time.Second
+)
+
+// podmanCommandContainer containerizes a command's execution using a Podman container.
+// between containers.
+type podmanCommandContainer struct {
+	image       string
+	hostRootDir string
+	// workDir is the path to the workspace directory mounted to the container.
+	workDir string
+}
+
+func NewPodmanCommandContainer(image, hostRootDir string) container.CommandContainer {
+	return &podmanCommandContainer{
+		image:       image,
+		hostRootDir: hostRootDir,
+	}
+}
+
+func (c *podmanCommandContainer) Run(ctx context.Context, command *repb.Command, workDir string, creds container.PullCredentials) *interfaces.CommandResult {
+	result := &interfaces.CommandResult{
+		CommandDebugString: fmt.Sprintf("(podman) %s", command.GetArguments()),
+		ExitCode:           commandutil.NoExitCode,
+	}
+	containerName, err := generateContainerName()
+	if err != nil {
+		result.Error = status.UnavailableErrorf("failed to generate podman container name: %s", err)
+		return result
+	}
+	if err := c.PullImageIfNecessary(ctx); err != nil {
```

Comment:
See comments on `c.PullImageIfNecessary` below -- if you implement `PullImage` and `IsImageCached` then you can switch this to `container.PullImageIfNecessary(ctx, env, imageCacheAuthenticator, ...)`. The `PullImageIfNecessary` util func takes care of auth for locally cached images.

(Using that func would require plumbing the imageCacheAuthenticator through to `podmanCommandContainer`.)

Reply:
I will plumbing imageCacheAuthenticator in a later PR.